lib/facter/es_facts.rb

```diff
@@ -50,6 +50,8 @@ def self.run
 
     if File.readable?("#{dir_prefix}/elasticsearch.yml")
       config_data = YAML.load_file("#{dir_prefix}/elasticsearch.yml")
+      return unless config_data
```

Comment:
I did some testing. YAML.load_file() fails if the file doesn't exist. we already guard that case with the File.readable?. YAML.load_file returns false if the yml is broken/fails.